src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -295,7 +295,20 @@ public static bool TryDecimal(FormattingInfo formatInfo, IRContext irContext, Fo
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-text
         public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
-        {
+        {
+            if (args.Length == 1 && (args[0] is BlankValue || (args[0] is UntypedObjectValue untypedObjectValue && untypedObjectValue.Impl.Type == FormulaType.Blank)))
+            {
+                return new BlankValue(irContext);

```

Comment:
Can you add a comment explaining the logic? Something like "// When used as a pure conversion function (single argument, no format string), Text() propagates null values"

Reply:
Add comment. thanks